crates/core/tedge_mapper/src/c8y/tests.rs

```diff
@@ -929,37 +927,31 @@ async fn start_c8y_mapper(mqtt_port: u16) -> Result<JoinHandle<()>, anyhow::Erro
     let mapper_task = tokio::spawn(async move {
         let _ = mapper.run().await;
     });
-    Ok(mapper_task)
+    Ok((_temp_dir, mapper_task))
```

Comment:
temp_dir needs to be returned so it does not go out of scope... how did this even work before??